src/main/java/uk/gov/gchq/magmacore/database/MagmaCoreObjectDatabase.java

```diff
@@ -51,7 +52,7 @@ public Thing get(final IRI iri) {
      */
     @Override
     public void create(final Thing object) {
-        final IRI iri = object.getIri();
+        final IRI iri = new IRI(object.getId());
```

Comment:
MC creates it's objects with IRI strings as the `Thing` IDs, so this should create a valid full IRI. If the IRI fails to parse then there will be an exception which indicates an issue elsewhere in the program since an object was somehow created without an IRI as its ID value.

Reply:
I see, I was confused because I didn't realise the object database was also using the RDF variant of HQDM. Do you think we should change this to use hqdm-core objects instead at some point? Or is that not a use-case we'll support?

Reply:
MagmaCore is currently RDF-specific so it seems reasonable to use IRIs here. A future change might be to make MagmaCore independent of RDF throughout, but I can't see an easy way to do that at the moment so it's probably best left until it's really needed.